jsiSIE/jsiSIE/SieDataItem.cs

```diff
@@ -165,18 +166,41 @@ internal string GetString(int field)
             if (Data.Count <= field) return null;
 
             var foo = Data[field].Trim();
-
+            //if date string empty
             if (string.IsNullOrEmpty(foo)) return null;
 
-            if (foo.Length != 8) Document.Callbacks.CallbackException(new SieDateException(foo + " is not a valid date"));
+            //if date string has strange lenght
+            if (foo.Length != 8 && foo.Length!= 10)
+            {
+                Document.Callbacks.CallbackException(new SieDateException($"{foo} is not a valid date (raw data: '{this.RawData}')"));
+                return null;
+            }
+
+            //Let's try to parse the date.
+            string[] styles = { "yyyy-MM-dd", "yyyyMMdd" };
```

Comment:
Since this is in violation of the SIE standard I can't let it slip past a user without notification.

With that said I; I like the use of format to parse the date so I will add date-formats as a public property of SieDocument and use most of your code.